### PR TITLE
Fixes to the session captures timeline data

### PR DESCRIPTION
**Pull request by mihow**
- Fixes the issue where clicking on a "spike" (interval) activates the _first_ capture in the interval, which may have no detections and feels confusing. Now when clicking on a spike the capture with the most detections is activated instead of the first capture in the interval.
- Fixes the num captures displayed per interval. Now uses the mode average of detections from captures within the interval (ignoring zeros). Before it was a sum of all detections in the interval, which felt misleading. 
- Fixes the issue where capture ticks were not visible if the number of detections is high. Now the captures Y axis has its own range (centered around its average) that is separate from the range of the detections y axis.
- Added vertical grid lines on the hour marks! Just a quick way to do it until we can add labels in #482 

Before:
image
After:
image

Before:
image
After:
image

### CODE REVIEW

File: ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx [hunk 1]
```diff
@@ -69,14 +90,30 @@ export const ActivityPlot = ({
...
+              showline: false,
+              showgrid: true,
+              griddash: 'dot',
+              gridwidth: 1,
+              gridcolor: gridLineColor,
+              showticklabels: false,
+              zeroline: false,
               fixedrange: true,
-              range: [session.startDate, session.endDate],
+              range: [new Date(session.startDate), new Date(session.endDate)],
```
Comment (annavik) [inline ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx:115]: `session.startDate` is already a date, any reason we are creating a new date? :)

**Review by annavik — COMMENTED** [review]
Skärmavbild 2024-08-08 kl  17 34 10

When number of detections are 0, can we make the y axis still start at 0?

File: ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx [hunk 2]
```diff
@@ -69,14 +90,30 @@ export const ActivityPlot = ({
...
             },
-            xaxis: {
-              showline: true,
+            yaxis2: {
               showgrid: false,
               showticklabels: false,
               zeroline: false,
+              rangemode: 'nonnegative',
+              fixedrange: true,
+              range: [0, session.detectionsMaxCount],
```
Comment (annavik) [inline ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx:102]: For example (regarding comment above):

```suggestion
              range: [0, session.detectionsMaxCount ?? yAxisMax],
```

File: ui/src/data-services/models/timeline-tick.ts [hunk 3]
```diff
@@ -50,6 +70,6 @@ export class TimelineTick {
       },
     })
 
-    return `${timespanString}<br>Captures: ${this.numCaptures}<br>Detections: ${this.numDetections}`
+    return `${timespanString}<br>Captures: ${this.numCaptures}<br>Avg. Detections: ${this.avgDetections}`
```
Comment (annavik) [inline ui/src/data-services/models/timeline-tick.ts:73]: Nit (just to be consistent with other labels):
```suggestion
    return `${timespanString}<br>Captures: ${this.numCaptures}<br>Avg. detections: ${this.avgDetections}`
```

**Review by annavik — APPROVED** [review]
Thank for fixing this, it was bugging me as well, especially clicking a spike and the sometimes outshined capture graph. I think it looks great, just that case when no detections I thought looked a bit strange, but no biggie!

Pushed a tiny fix to format numbers in the capture picker.

File: ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx (hunk 1 above)
Comment (mihow) [reply ui/src/pages/session-details/playback/activity-plot/activity-plot.tsx:115]: You are right, I was experimenting with date formats was trying different approaches. Reverting!

**Comment by mihow** [comment]
> Skärmavbild 2024-08-08 kl 17 34 10
> When number of detections is 0, can we make the y axis still start at 0?

Fixed!

**Pull request merged**